recipes/metanetx-assets/meta.yaml

```diff
@@ -19,20 +19,20 @@ build:
 requirements:
   host:
     - pip
-    - python >=3.8
+    - python =3.8
     - setuptools
   run:
-    - click ~=7.0
-    - click-log ~=0.3
-    - cobra-component-models ~=0.3
-    - depinfo ~=1.5
-    - httpx ~=0.11
-    - pandas ~=0.25
-    - pydantic ~=1.4
-    - pyparsing ~=2.4
-    - python >=3.8
-    - SQLAlchemy ~=1.3
-    - tqdm ~=4.0
+    - click >=7.0,<8
+    - click-log >=0.3,<1
+    - cobra-component-models >=0.3,<1
+    - depinfo >=1.5,<2
+    - httpx >=0.11,<1
+    - pandas >=0.25,<1
+    - pydantic >=1.4,<2
+    - pyparsing >=2.4,<3
+    - python =3.8
+    - SQLAlchemy >=1.3,<2
```

Comment:
So conda has fuzzy matching. You can use `=` instead of the joint constraint.

Reply:
From the docs that you linked above:

> EXAMPLE: python=3.4 is translated to python 3.4*.

which is not what I want. I want to allow all minor version increases (hoping that they will not break the API) up until the next major version release. This is what I was aiming for with `~=` specified in [PEP440](https://www.python.org/dev/peps/pep-0440/#compatible-release).

Reply:
You could also use the jinja2 pinning statements. See pin_compatible here: https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#extra-jinja2-functions

I am trying to find ways to make this easier to maintain.

An issue on PEP440 for conda btw: https://github.com/conda/conda/issues/4948

Reply:
> `pin_compatible('package_name', min_pin='x.x.x.x.x.x', max_pin='x', lower_bound=None, upper_bound=None)`: To be used as pin in run and/or test requirements. Takes package name argument. Looks up compatibility of named package installed in the build environment and writes compatible range pin for run and/or test requirements. Defaults to a semver-based assumption: `package_name >=(current version),<(next major version)`.

Since I only have `run` dependencies and no `build` dependencies, would this work?

Reply:
It should. I think "build" environment is interpreted loosely here.

Reply:
I'm wondering if that is really so much more maintainable but will give it a shot.

Reply:
W/e works for you!